src/backend/chat/commands/command-handler.ts

```diff
@@ -0,0 +1,640 @@
+
+        const userCmd: UserCommand = {
+            trigger: trigger,
+            args: args,
+            commandSender: sender,
+            senderRoles: senderRoles ?? []
+        };
+
+        if (!command.scanWholeMessage &&
+            !command.triggerIsRegex &&
+            userCmd.args.length > 0 &&
+            command.subCommands?.length > 0) {
+
+            for (const subcmd of command.subCommands) {
+                if (subcmd.active === false && command.type !== "system") {
+                    continue;
+                }
+                if (subcmd.regex) {
+                    const regex = new RegExp(`^${subcmd.arg}$`, "gi");
+                    if (regex.test(userCmd.args[0])) {
+                        userCmd.triggeredSubcmd = subcmd;
+                        break;
+                    }
+                } else {
+                    if (subcmd.arg.toLowerCase() === userCmd.args[0].toLowerCase()) {
+                        userCmd.triggeredSubcmd = subcmd;
+                        break;
+                    }
+                }
+            }
+
+            if (command.type !== "system" && userCmd.triggeredSubcmd == null) {
+                if (command.fallbackSubcommand == null || !command.fallbackSubcommand.active) {
+                    userCmd.isInvalidSubcommandTrigger = true;
+                } else {
+                    userCmd.triggeredSubcmd = command.fallbackSubcommand;
+                }
+            }
+
+            if (userCmd.triggeredSubcmd != null) {
+                userCmd.triggeredArg = userCmd.triggeredSubcmd.arg;
+                userCmd.subcommandId = userCmd.triggeredSubcmd.id;
+            }
+        }
+
+        return userCmd;
+    }
+
+    private fireCommand(
+        command: CommandDefinition,
+        userCmd: UserCommand,
+        firebotChatMessage: FirebotChatMessage,
+        commandSender: string,
+        isManual = false
+    ): void {
+        if (command == null) {
+            return;
+        }
+        if (commandSender == null) {
+            commandSender = accountAccess.getAccounts().streamer.username;
+        }
+
+        logger.info(`Checking command type... ${command.type}`);
+
+        if (command.type === "system") {
+            logger.info("Executing system command");
+            //get system command from manager
+            const cmdDef = commandManager.getSystemCommandById(command.id);
+
+            const commandOptions = {};
+            if (command.options != null) {
+                for (const optionName of Object.keys(command.options)) {
+                    const option = command.options[optionName];
+                    if (option) {
+                        let value = option.value;
+                        if (value == null) {
+                            value = option.default;
+                        }
+                        commandOptions[optionName] = value;
```

Comment:
I know this is preexisting stuff but for cleanups sake while we are in here, this could be modernized/simplified to:
```ts
commandOptions[optionName] = option.value ?? option.default
```